app/models/spree/product_set.rb

```diff
@@ -28,9 +28,35 @@ def update_attributes(attributes)
     if found_model.nil?
       @klass.new(attributes).save unless @reject_if.andand.call(attributes)
     else
-      ( attributes.except(:id, :variants_attributes, :master_attributes).present? ? found_model.update_attributes(attributes.except(:id, :variants_attributes, :master_attributes)) : true) and
-      (attributes[:variants_attributes] ? update_variants_attributes(found_model, attributes[:variants_attributes]) : true ) and
-      (attributes[:master_attributes] ? update_variant(found_model, attributes[:master_attributes]) : true )
+      update_product_only_attributes(found_model, attributes) &&
+        update_product_variants(found_model, attributes) &&
+        update_product_master(found_model, attributes)
+    end
+  end
+
+  def update_product_only_attributes(product, attributes)
+    if attributes.except(:id, :variants_attributes, :master_attributes).present?
+      product.update_attributes(
+        attributes.except(:id, :variants_attributes, :master_attributes)
+      )
+    else
+      true
+    end
+  end
```

Comment:
dont you prefer the early exit style for these if clauses?

  ```
return true unless attributes.except(:id, :variants_attributes, :master_attributes).present?
product.update_attributes(
   attributes.except(:id, :variants_attributes, :master_attributes)
)  
```

Reply:
I do. I just want to touch them the least :trollface:. Will do